File: src/main/java/com/openlattice/util/Hashcodes.java
package com.openlattice.util;

public class Hashcodes {

    private Hashcodes() {
        /* No-Op */
    }

    public static int generate( Object a, Object b ) {
        int result = 31 + (a == null ? 0 : a.hashCode());
        return 31 * result + ( b == null ? 0 : b.hashCode());
    }

    public static int generate( Object a, Object b, Object c ) {
        return 31 * generate(a, b) + ( c == null ? 0 : c.hashCode());
    }

    public static int generate( Object a, Object b, Object c, Object d) {
        return 31 * generate(a, b, c) + ( d == null ? 0 : d.hashCode());
    }

    public static int generate( Object a, Object b, Object c, Object d, Object e) {
        return 31 * generate(a, b, c, d) + ( e == null ? 0 : e.hashCode());
    }

    public static int generate( Object a, Object b, Object c, Object d, Object e, Object f) {
        return 31 * generate(a, b, c, d, e) + ( f == null ? 0 : f.hashCode());
    }
}

File: src/main/kotlin/com/openlattice/util/Hashcodes.kt
package com.openlattice.util

class Hashcodes {

    companion object {
        @JvmStatic
        fun generate(a: Any?, b: Any?): Int {
            val result = 31 * 1 + if (a == null) 0 else a.hashCode()
            return 31 * result + if (b == null) 0 else b.hashCode()
        }

        @JvmStatic
        fun generate(a: Any?, b: Any?, c: Any?): Int {
            return 31 * generate( a, b ) + if (c == null) 0 else c.hashCode()
        }

        @JvmStatic
        fun generate(a: Any?, b: Any?, c: Any?, d: Any?): Int {
            return 31 * generate(a, b, c) + if (d == null) 0 else d.hashCode()
        }

        @JvmStatic
        fun generate(a: Any?, b: Any?, c: Any?, d: Any?, e: Any?): Int {
            return 31 * generate( a, b, c, d ) + if (e == null) 0 else e.hashCode()
        }

        @JvmStatic
        fun generate(a: Any?, b: Any?, c: Any?, d: Any?, e: Any?, f: Any?): Int {
            return 31 * generate( a, b, c, d, e ) + if (f == null) 0 else f.hashCode()
        }
    }
}